types/audit-details.d.ts

```diff
@@ -94,60 +93,36 @@ declare global {
          */
         key: string|null;
         /** Readable text label of the field. */
-        text: string;
+        label: string;
         /**
          * The data format of the column of values being described. Usually
          * those values will be primitives rendered as this type, but the values
          * could also be objects with their own type to override this field.
          */
-        itemType: ItemValueTypes;
+        valueType: ItemValueType;
         /**
          * Optional - defines an inner table of values that correspond to this column.
          * Key is required - if other properties are not provided, the value for the heading is used.
          */
-        subRows?: {key: string, itemType?: ItemValueTypes, displayUnit?: string, granularity?: number};
+        subRows?: {key: string, valueType?: ItemValueType, displayUnit?: string, granularity?: number};
 
         displayUnit?: string;
         granularity?: number;
       }
 
-      export type TableItem = {
+      export interface TableItem {
         debugData?: DebugData;
-        [p: string]: undefined | Value | Value[];
-      }
-
-      export interface OpportunityColumnHeading {
-        /**
-         * The name of the property within items being described.
-         * If null, subRows must be defined, and the first sub-row will be empty.
-         */
-        key: string|null;
-        /** Readable text label of the field. */
-        label: string;
-        /**
-         * The data format of the column of values being described. Usually
-         * those values will be primitives rendered as this type, but the values
-         * could also be objects with their own type to override this field.
-         */
-        valueType: ItemValueTypes;
-        /**
-         * Optional - defines an inner table of values that correspond to this column.
-         * Key is required - if other properties are not provided, the value for the heading is used.
-         */
-        subRows?: {key: string, valueType?: ItemValueTypes, displayUnit?: string, granularity?: number};
-
-        // NOTE: not used by opportunity details, but used in the renderer until table/opportunity unification.
-        displayUnit?: string;
-        granularity?: number;
+        [p: string]: undefined | ItemValue | ItemValue[];
       }
 
-      export interface OpportunityItem {
+      /** A more specific table element used for `opportunity` tables. */
+      export interface OpportunityItem extends TableItem {
```

Comment:
`OpportunityItem` was already a `TableItem` in tsc's eyes (since it checks using structural typing), this just makes the relationship more explicit. `OpportunityItem` exists to let opportunity audits know what properties to include on their table items, while the more generic `TableItem` is what `details-renderer` knows how to render (regardless of specific keys).